### PR TITLE
vscode-extensions: detect vsix sources and generate packages automatically

### PR DESCRIPTION
**Pull request by danielphan2003**
Looking at PR #348, I notice that we can improve how `vscode-extensions` handles generated vsix source in [pkgs/sources.toml](https://github.com/danielphan2003/devos/tree/vscode-extensions).

This PR aims to [`mkVscodeExtension`](https://github.com/danielphan2003/devos/blob/vscode-extensions/pkgs/misc/vscode-extensions/vscode-utils.nix#L46) with auto-generated meta attributes and passthru values such as `meta.homepage`, `meta.downloadPage`, `meta.changelog`, `meta.license`, `meta.maintainers`, `meta.platforms`, `{drv}Phase`, etc.

Therefore, the [following](https://github.com/berberman/nvfetcher/blob/master/nvfetcher_example.toml#L48-L51) code snippet can be simplified into:
```
[vscode-LiveServer]
src.openvsx = "ritwickdey.LiveServer"
fetch.openvsx = "ritwickdey.LiveServer"
passthru = { license = "mit", homepage = "https://github.com/ritwickdey/vscode-live-server", description = "Launch a development local Server with live reload feature for static & dynamic pages" }
```

The `passthru` can be generated through `bud` script, although I will need to test more of this later.

### CODE REVIEW

File: pkgs/misc/vscode-extensions/vscode-utils.nix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}:
+
+with lib;
+
+let
+  isVsix = hasSuffix ".vsix";
+
+  isVsixPackage = hasSuffix ".VSIXPackage";
+
+  isVscodeExt = name: (isVsix name) || (isVsixPackage name);
```
Comment (blaggacao) [inline pkgs/misc/vscode-extensions/vscode-utils.nix:14]: Nice!

File: pkgs/misc/vscode-extensions/default.nix [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{ lib, vscode-utils, sources }:
+let
+  inherit (vscode-utils) isVscodeExt mkVscodeExtensions;
```
Comment (blaggacao) [inline pkgs/misc/vscode-extensions/default.nix:3]: Those look like convenience library functions that we might want to discuss adding in [`digga`](https://github.com/divnix/digga)?

Comment (blaggacao) [reply pkgs/misc/vscode-extensions/default.nix:3]: or alternatively a `divnix/vsext` (to decouple release cycles)!!!

**Comment by blaggacao** [comment]
IMO, also `bud` complements can exquisitely live in a `divnix/vsext`.

**Comment by danielphan2003** [comment]
Superseeded by https://github.com/divnix/devos-ext-lib

**Pull request closed without merging**